src/Analysis/Ast/Impl/Analyzer/PythonAnalyzerEntry.cs

```diff
@@ -139,9 +140,10 @@ public Task<IDocumentAnalysis> GetAnalysisAsync(CancellationToken cancellationTo
                 _analysisDependencies = null;
                 UpdateAnalysisTcs(version);
                 _previousAnalysis = analysis;
+                tcs = _analysisTcs;
             }
 
-            _analysisTcs.TrySetResult(analysis);
+            tcs.TrySetResult(analysis);
```

Comment:
`_analysisTcs` can be changed by another call of `UpdateAnalysisTcs` with a newer version. Shouldn't affect the old one.